frontends/PyCDE/src/pycde/devicedb.py

```diff
@@ -119,3 +119,9 @@ def get_instance_at(self, loc: PhysLocation) -> object:
       return None
     # TODO: resolve instance and return it.
     return inst
+
+  def reserve_location(self, loc: PhysLocation, entity: msft.EntityExternOp):
+    ref = FlatSymbolRefAttr.get(entity.sym_name.value)
+    path = ArrayAttr.get([ref])
+    subpath = ""
```

Comment:
Are there constraints on what we can put in the path and subpath here? Would it be allowable for me to create an EntityExternOp for a module and then use the path or subpath fields to store additional data about which op in a module is being placed? This would be with the understanding that CIRCT would ignore that anyway, but it could be read when walking the placements, etc.

Reply:
Here the path needs to point to the entity by its symbol name so the later passes can find the entity (and know to ignore it). In this case, the subpath isn't being used, but I don't know if you would want to use it for extra information. When you create the entity, you can attach arbitrary data in the `metadata` attribute, which will be available on the operation when the placements are walked, etc.

Reply:
That makes sense. Thanks!